### PR TITLE
Fix ComponentDocumentDesigner references in DesignerAttribute to include right base type

### LINKED ISSUE #31428: IComponent is missing designer attributes
In particular, `IComponent` is missing the following [two attributes](https://referencesource.microsoft.com/#System/compmod/system/componentmodel/IComponent.cs,34), which were defined on .NET Framework:

```C#
[Designer("System.ComponentModel.Design.ComponentDesigner, " + AssemblyRef.SystemDesign, typeof(IDesigner))]
[Designer("System.Windows.Forms.Design.ComponentDocumentDesigner, " + AssemblyRef.SystemDesign, typeof(IRootDesigner))]
```

This affects both designer and runtime `IDesignerHost` scenarios. cc @ericstj. 

While we've worked around this in our designer by hard-coding knowledge of `IComponent`'s missing attributes in our own `IDesignerHost` implementation, third-parties would be hard-pressed to do the same for runtime scenarios where the designer is hosted in an application at runtime. An example of such a scenario is DevExpress's [Reporting](https://docs.devexpress.com/XtraReports/2162/reporting) library where they allow a report designer based on CodeDom serialization to embedded directly into an end user application. This is a very meaningful for certain line-of-business apps.

### PR DESCRIPTION
**Pull request by safern**
Fixes: https://github.com/dotnet/runtime/issues/31428

### CODE REVIEW

File: src/libraries/System.ComponentModel.Primitives/ref/System.ComponentModel.Primitives.cs [hunk 1]
```diff
@@ -181,7 +181,7 @@ public sealed partial class EventHandlerList : System.IDisposable
         public void RemoveHandler(object key, System.Delegate? value) { }
     }
     [System.ComponentModel.DesignerAttribute("System.ComponentModel.Design.ComponentDesigner, System.Design, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a")]
-    [System.ComponentModel.DesignerAttribute("System.Windows.Forms.Design.ComponentDocumentDesigner, System.Design, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a")]
+    [System.ComponentModel.DesignerAttribute("System.Windows.Forms.Design.ComponentDocumentDesigner, System.Design, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a", typeof(System.ComponentModel.Design.IRootDesigner))]
```
Comment (ericstj) [inline src/libraries/System.ComponentModel.Primitives/ref/System.ComponentModel.Primitives.cs:184]: You'll need the fully qualified name for this one.  It lives in TypeConverter on core.  It lived in System on .NETFramework.  Let's use the .NETFramework identity.

**Comment by ericstj** [comment]
We should make sure to test this. @DustinCampbell suggested

> The repro would be to create an IComponent instance and call TypeDescriptor.CreateDesigner(...) with the base type as typeof(IDesigner) and then again with typeof(IRootDesigner). Today, both calls will return ComponentDesigner, but the second should return ComponentDocumentDesigner.

We'd need changes to our test infra to do that in a our unit tests, since it'd require a windows desktop shared framework, but we should make sure we at least manually verify this.

**Comment by safern** [comment]
Sounds good. I'll test this locally.

File: src/libraries/System.ComponentModel.Primitives/ref/System.ComponentModel.Primitives.cs (hunk 1 above)
Comment (safern) [reply src/libraries/System.ComponentModel.Primitives/ref/System.ComponentModel.Primitives.cs]: Good catch. I fixed this and also fixed another instance of this in another DesignerAttribute that used this designer.

File: src/libraries/shims/ApiCompatBaseline.PreviousNetCoreApp.txt [hunk 2]
```diff
@@ -1,8 +1,18 @@
-Compat issues with assembly System.Diagnostics.Process:
+Compat issues with assembly netstandard:
+CannotChangeAttribute : Attribute 'System.ComponentModel.DesignerAttribute' on 'System.ComponentModel.IComponent' changed from '[DesignerAttribute("System.ComponentModel.Design.ComponentDesigner, System.Design, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a")]' in the contract to '[DesignerAttribute("System.ComponentModel.Design.ComponentDesigner, System.Design, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a")]' in the implementation.
```
Comment (ericstj) [inline src/libraries/shims/ApiCompatBaseline.PreviousNetCoreApp.txt:2]: I see that we're catching this now as a diff with past release.  Does API compat flag the original issue as a diff with netfx?  If so are there any other similar diffs?

**Review by ericstj — APPROVED** [review]

File: src/libraries/shims/ApiCompatBaseline.PreviousNetCoreApp.txt (hunk 2 above)
Comment (safern) [reply src/libraries/shims/ApiCompatBaseline.PreviousNetCoreApp.txt:2]: No it didn't catch it. I don't know why. Based on reading these errors, it seems like APICompat is not catching the right "Diff" the diff is in the `DesignerAttribute` that specifies a `ComponentDocumentDesigner`.  

This is the only diff I get on my local build and on CI.

**Comment by safern** [comment]
I tested this locally using the local runtime pack we build in dotnet/runtime and I can see the issue no longer repros. I reproed and then applied the fix and it is no longer showing up. Merging this.

**Pull request merged**